lessons/PowerShell/http/http-lesson-powershell.md

```diff
@@ -0,0 +1,267 @@
+6. Select `Function` for the AccessRights.
+    > 🔎 **Observation** - Now a new Azure Functions project is being generated. Once it's done, look at the files in the project. You will see the following:
+
+    |File|Description
+    |-|-
+    |host.json|Contains [global configuration options](https://docs.microsoft.com/en-us/azure/azure-functions/functions-host-json) for all the functions in a function app.
+    |profile.ps1| A profile file, like you would have in a local PowerShell prompt. Here you can store all cmdlets that need to be executed when the function [cold starts](https://azure.microsoft.com/en-us/blog/understanding-serverless-cold-start/).
+    |proxies.json| Can be used if the Function app is part of a larger API, is not used for standalone Functions. [More information](https://docs.microsoft.com/en-us/azure/azure-functions/functions-proxies)
+    |requirements.psd1| Define modules in the PowerShell gallery that need to be loaded when the function starts.
+    |local.settings.json|Contains [app settings and connectionstrings](https://docs.microsoft.com/en-us/azure/azure-functions/functions-run-local?tabs=windows%2Ccsharp%2Cbash#local-settings-file) for local development.
+    |**HelloWorldHttpTrigger**|
+    |function.json| The trigger and binding settings of the Function
+    |run.ps1| The actual PowerShell script that will run when the function is triggered
+    |sample.dat| and example body input for the function http trigger.
+
+    > ❔ **Question** - Review the generated HTTPTrigger function. What is it doing?
+7. Start the Function App by pressing `F5`.
+    > 🔎 **Observation** - Eventually you should see an HTTP endpoint in the output.
+8. Now call the function by opening a PowerShell prompt outside of Visual Studio Code and using Invoke-RestMethod:
+
+    ```PowerShell
+     Invoke-RestMethod http://localhost:7071/api/HelloWorldHttpTrigger?name=YourName
+    ```
+
+    >📝 **Tip** - In Invoke-RestMethod, you can choose not to define a `Method`. If you don't, it will always use `GET`.
+
+    > 🔎 **Observation** - We use Invoke-RestMethod, but you could also use Curl or just your web browser!
+
+    > ❔ **Question** - What is the result of the function? Is it what you expected?
+
+    > ❔ **Question** - What happens when you don't supply a value for the name?
+
+## 2. Changing the template for GET requests
+
+Let's change the template to find out what parameters can be changed.
+Depending on the trigger, arguments can be added/removed and parameter types can be changed.
+Start with only allowing GET requests.
+
+### Steps
+
+1. Open function.json. In the httpTrigger binding, the `methods` are listed. Remove `"post"`. Don't forget to remove the comma after `"get"`. Now the function can only be triggered by a GET request.
+
+    > 🔎 **Observation** - You'll notice that this change breaks the code inside the function. This is because the `HttpRequestMessage` type has different properties and methods than the `HttpRequest` type.
```

Comment:
I didn't test the exercise myself just yet, but is this observation correct here? I don't see any mention of changing the request type in step 1 (?).

Reply:
The request type was changed by changing the json, but that did not break the function. Removed the observation